tests/system/test_cluster/test_agent_groups/test_group_sync_status.py

```diff
@@ -0,0 +1,185 @@
+'''
+copyright: Copyright (C) 2015-2023, Wazuh Inc.
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+type: system
+brief: Wazuh manager handles agent groups. 
+       If a group is deleted from a master cluster, there will be an instance where the agents require a resynchronization (syncreq). 
+       If the group is deleted from a worker cluster, the cluster master will take care of reestablishing the group structure without the need for resynchronization.
+       This test suite tests the correct functioning of the mentioned use case.
+
+tier: 0
+modules:
+    - enrollment
+components:
+    - manager
+    - agent
+daemons:
+    - wazuh-authd
+    - wazuh-agentd
+os_platform:
+    - linux
+os_version:
+    - Debian Buster
+references:
+    - https://documentation.wazuh.com/current/user-manual/registering/agent-enrollment.html
+'''
+
+import os
+import pytest
+import time
+from wazuh_testing.tools import WAZUH_PATH, WAZUH_LOGS_PATH
+from wazuh_testing.tools.file import read_file, read_yaml
+from wazuh_testing.tools.system import HostManager
+from system import clean_cluster_logs
+
+pytestmark = [pytest.mark.cluster, pytest.mark.enrollment_cluster_env]
+
+testinfra_hosts = ['wazuh-master', 'wazuh-worker1','wazuh-worker2']
+groups = ['group_master', 'group_worker1', 'group_worker2']
+agents = ['wazuh-agent1', 'wazuh-agent2']
+workers = ['wazuh-worker1', 'wazuh-worker2']
+groups_created = []
+first_time_check = "synced"
+second_time_check = "synced"
+network = {}
+client_keys = {}
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+test_cases_yaml = read_yaml(os.path.join(local_path, 'data/test_group_sync_cases.yml'))
+wdb_query = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'script/wdb-query.py')
+
+@pytest.fixture
+def delete_logs():
```

Comment:
Check if we can use `clean_environment()`

Reply:
Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143